### PR TITLE
Granthamblin0/tces 45 clt client table

### PR DESCRIPTION
**Pull request by granthamblin**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Issue #

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# Checklist

Please delete options that are not applicable.

- [x] My code follows the style guidelines of this project (required)
- [x] I have performed a self-review of my code (required)
- [x] I have linted and formatted my code (required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: client/src/App.jsx
Comment (kevinle623) [inline client/src/App.jsx]: can this be switched back to false?

File: client/src/components/client-dashboard-component/ClientTable.jsx
Comment (kevinle623) [inline client/src/components/client-dashboard-component/ClientTable.jsx]: can we move columns into the function

**Review by kevinle623 — COMMENTED** [review]
Nice work Grant! Left some comments, just small adjustments but overall looks good

File: client/src/components/client-dashboard-component/FilterCard.jsx
Comment (DanielDervishi) [inline client/src/components/client-dashboard-component/FilterCard.jsx]: We forgot to ask Jake to remove it but Jordan and I decided a while ago to get rid of distinguishing between all closed clients and ones with action needed. Just having the "Closed" checkbox without the additional options should be sufficient here.
Screenshot 2024-01-26 at 12 28 38 PM

File: client/src/components/client-dashboard-component/FilterCard.jsx
Comment (DanielDervishi) [inline client/src/components/client-dashboard-component/FilterCard.jsx]: I think the whole date filtering might need a bit of a rework for the sake of uniformity with the other tables. I'd recommend taking a look at the job leads or employer tables as a reference.

File: client/src/components/client-dashboard-component/FilterCard.jsx
Comment (DanielDervishi) [inline client/src/components/client-dashboard-component/FilterCard.jsx]: Add a button for applying filters as well
Screenshot 2024-01-26 at 12 31 18 PM

File: client/src/components/client-dashboard-component/ClientTable.jsx
Comment (DanielDervishi) [inline client/src/components/client-dashboard-component/ClientTable.jsx]: I recommend taking a look at how Kevin or Emily have styled the table (Job leads and Employer tables) and changing your table to closer match theirs.  Some particular things to focus on are:
- the positioning of the table relative to the filter component
- the table border
- the outline of the owner column
Further I'd take a look at using https://mui.com/material-ui/react-chip/ for the activity status instead of manually trying to match the designs.
Screenshot 2024-01-26 at 12 35 05 PM

File: client/src/components/client-dashboard-component/ClientTable.jsx
Comment (DanielDervishi) [inline client/src/components/client-dashboard-component/ClientTable.jsx]: atm when a client hovers over the title of a column or clicks on one there seems to be some indication of ordering present (up arrow or down arrow) beside the title of that column. I think Kevin's job leads table doesn't have that so you can definitely use it as a reference for removing this. Screenshot 2024-01-26 at 12 41 55 PM

**Review by DanielDervishi — COMMENTED** [review]
Great work Grant! I like how you've separated the components here. I just left a few comments mainly surrounding unifying the design of the three table pages (job leads, clients and employers)

File: client/src/components/client-dashboard-component/FilterCard.jsx
Comment (jordanjanakievski) [reply client/src/components/client-dashboard-component/FilterCard.jsx]: If you would like, you can copy the way Kevin has those buttons implemented on the job-leads-table. Otherwise these button will be cleaned up in a small ticket later on

File: client/src/components/client-dashboard-component/client-dashboard-table/ClientTable.jsx
Comment (DanielDervishi) [inline client/src/components/client-dashboard-component/client-dashboard-table/ClientTable.jsx:88]: The right page navigation button is currently being cut off by the table container
Screenshot 2024-02-02 at 5 37 40 PM

File: client/package.json
Comment (DanielDervishi) [inline client/package.json:15]: Seems that there were two different designs created last year for the client table page with the main difference being the introduction of a back button & title. It would be great if we could do something like this (the back button and title) since every other table page follows this same pattern
Screenshot 2024-02-02 at 5 42 11 PM

**Review by DanielDervishi — COMMENTED** [review]
Great work Grant! Should be good to merge after these small comments are handled

**Comment by jordanjanakievski** [comment]
Just a reminder note, App.jsx will need to be reverted before merge

File: client/src/components/client-dashboard-component/index.jsx
Comment (jordanjanakievski) [inline client/src/components/client-dashboard-component/index.jsx]: Remove the padding and set all the margins to 20px

**Review by DanielDervishi — DISMISSED** [review]
LGTM!

**Review by kevinle623 — DISMISSED** [review]

**Review by jordanjanakievski — DISMISSED** [review]
Fix app js

**Review by DanielDervishi — APPROVED** [review]

**Review by kevinle623 — APPROVED** [review]

**Pull request merged**
